admin/javascript/SSNotice.js

```diff
@@ -0,0 +1,205 @@
+/**
+* File: SSNotice.js
+*
+* Replace the old statusMessage/errorMessage
+* with a custom library: ss.notice
+*
+* Allows for concistent notices/message display throughout the CMS
+* via simple JavaScript function calls:
+* - ss.notice.message(text, [options]);
+* - ss.notice.success(text, [options]);
+* - ss.notice.warning(text, [details], [options]);
+* - ss.notice.error(text, [details], [options]);
+**/
+(function($, root) {
+
+	// local working version of the library object
+	var lib = {};
+
+	/**
+	 * Library settings
+	 */
+	lib.settings = {
+		parentSelector: '.cms-content-fields',
+		$notice: $('<p class="message"/>'),
```

Comment:
I understand this is creating a dom element through jQuery - but you're making some pretty fundamental decisions in the JS in regards to how they should be marked up.

Why should it always be a `<p>` and what if we want a different class (or extra ones)?

If you either have a JS template OR a hidden el which contains these items already, then you can pull them out a lot more nicely.

```js

$holderEl = $('.js-notice-holder');
$details = $holderEl.find('.js-error-details');

```

Then you're only relying on some abstract hooks to get the elements out and there is no mixing of concerns